qt/aqt/browser/card_info.py

```diff
@@ -63,15 +63,11 @@ def _setup_ui(self, card_id: CardId | None) -> None:
         layout.addWidget(buttons)
         qconnect(buttons.rejected, self.reject)
         self.setLayout(layout)
-
-        self.web.eval(
-            "const cardInfo = anki.cardInfo(document.getElementById('main'));"
-        )
         self.update_card(card_id)
 
     def update_card(self, card_id: CardId | None) -> None:
         self.web.eval(
-            f"cardInfo.then((c) => c.$set({{ cardId: {json.dumps(card_id)} }}));"
+            f"anki.cardInfoPromise.then((c) => c.$set({{ cardId: {json.dumps(card_id)} }}));"
```

Comment:
Since we've gone to the trouble of changing these references already, would it make sense to switch from a global to using our register-package functionality as well? Eg so calls like the above would become something like

```
require("anki/card-info").loaded.then(...)
```

I presumably that would give us a bit more flexibility in terms of renames and deprecations, and would be more discoverable than global variables. WDYT? Would there be disadvantages to doing it that way?

Reply:
Actually I had the same thought.
But rather than doing it per package, I wanted to expose one single package (in lib), that exposes, let's say, `uiLoaded` (or we could keep the iOS naming scheme: `uiDidLoad`).

My thought is that we're always going to have _one_ Svelte entry point per webview, so it would be easy to maintain going forward.

However this promise would not return any API, but rather `void`. To access the Editor API, we could expose it in an interface similar to `aqt.dialogs`:

```ts
const { uiDidLoad, uiComponents } = require("anki/ui")

await uiDidLoad;

// My intention is to rename `OldEditorAdapter.svelte` to
// `BaseEditor`, and have it contain all functionality shared between all
// editor instances.
const editor = uiComponents.get("baseEditor" /* or maybe a symbol */);
```

This would make things easier for add-ons and also safe going forward, however it'll create quite a bit boilerplate for `web.eval` calls from Python.

Reply:
No preferences on the name - on the Qt end where we often have hooks before and after some action, the awkward naming made some sense, but we don't necessarily need to keep it here.

A single promise for all screens sounds like it could work. If 'await ui, then load component' is a common use case, then perhaps we could add some helper routine to reduce the boilerplate in such cases. Symbols might be a better choice there, or some other approach that allows us to strongly type the input and output of uiComponents.get()